### PR TITLE
Various misc fixes

### PR DESCRIPTION
**Pull request by shuklaayush**

### CODE REVIEW

File: src/CitadelMinter.sol [hunk 1]
```diff
@@ -172,6 +172,16 @@ contract CitadelMinter is
...
     {
+        // Saves gas below if non-zero
+        uint256 cachedLockingBps = lockingBps;
+        uint256 cachedStakingBps = stakingBps;
+        uint256 cachedFundingBps = fundingBps;
+
+        require(
+            cachedLockingBps + cachedStakingBps + cachedFundingBps == MAX_BPS,
+            "CitadelMinter: Sum of propvalues must be 10000 bps"
+        );
```
Comment (sajanrajdev) [inline src/CitadelMinter.sol:183]: I don't think this require is necessary as the same check already happens within `setCitadelDistributionSplit` and this is the only way to alter these values.

**Review by sajanrajdev — COMMENTED** [review]
- General gas optimizations
- Adds the missing role management assigments to the CONTRACT_GOVERNANCE_ROLE
- Changed the tokenOutPrice variable name from the KnightingRound
- Adds the fee checks to the stakedCitadel contract
- Adapted tests according to changes

Other than the redundant BPS check, things LGTM

File: src/CitadelMinter.sol (hunk 1 above)
Comment (shuklaayush) [reply src/CitadelMinter.sol:183]: This is to ensure that `mintAndDistribute` is only called after `setCitadelDistributionSplit` has been called at least once

Comment (sajanrajdev) [reply src/CitadelMinter.sol:183]: I see, makes sense. LGTM

**Review by sajanrajdev — APPROVED** [review]
LGTM

**Pull request merged**